src/Components/Lobby.tsx

```diff
@@ -17,11 +17,11 @@ const Lobby = () => {
   const playerDispatch = usePlayerDispatch();
 
   const [playerName, setPlayerName] = useState("");
-  const [isHost, setIsHost] = useState(false);
   const [promptCode, setPromptCode] = useState(false);
   const [existingRoomCode, setExistingRoomCode] = useState("");
   const [failJoinRoomMessage, setFailJoinRoomMessage] = useState("");
   const [alert, setAlert] = useState<boolean>(false)
+  const [currentPlayer, setCurrentPlayer] = useState({})
```

Comment:
I noticed youre using currentPlayer just for the number value right?
we should be able to skip using a localState for this.
if you take a look at `usePlayerDocState` hook
it returns an object with these values. (dont worry about useMemo, its just to improve performance)
```
{players, setPlayers, player, setPlayer}
```

the player object should also have the number value
and it currently gets set once gameStarts (`src/Components/Board.tsx` line 40)

^ is still necessary as it updates player directions and abilities.
But you can update the player state earlier when creating/joining room.